daphne/src/lib.rs

```diff
@@ -236,6 +242,24 @@ pub struct ProblemDetails {
     pub detail: Option<String>,
 }
 
+/// DAP version used for a task.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub enum DapVersion {
+    Unknown,
+    Draft01,
+    Draft02,
+}
+
+impl DapVersion {
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            DapVersion::Unknown => "unknown",
+            DapVersion::Draft01 => "v01",
+            DapVersion::Draft02 => "v02",
+        }
+    }
```

Comment:
This method is used to convert a string into a `DapVersion`. The proper way to do this in Rust is to implement `From<&str> for DapVersion`.

Reply:
Thanks for the suggestion. I've implemented this.